flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopRecoverableWriter.java

```diff
@@ -121,24 +127,14 @@ public boolean supportsResume() {
 		return true;
 	}
 
-	@VisibleForTesting
-	static org.apache.hadoop.fs.Path generateStagingTempFilePath(
-			org.apache.hadoop.fs.FileSystem fs,
-			org.apache.hadoop.fs.Path targetFile) throws IOException {
-
+	private static org.apache.hadoop.fs.Path generateStagingTempFilePath(org.apache.hadoop.fs.Path targetFile) {
 		checkArgument(targetFile.isAbsolute(), "targetFile must be absolute");
 
 		final org.apache.hadoop.fs.Path parent = targetFile.getParent();
 		final String name = targetFile.getName();
 
 		checkArgument(parent != null, "targetFile must not be the root directory");
 
-		while (true) {
-			org.apache.hadoop.fs.Path candidate = new org.apache.hadoop.fs.Path(
-					parent, "." + name + ".inprogress." + UUID.randomUUID().toString());
-			if (!fs.exists(candidate)) {
-				return candidate;
-			}
-		}
+		return new org.apache.hadoop.fs.Path(parent, "." + name + ".inprogress");
```

Comment:
We should definitely wait for @StephanEwen to comment on this (as you already discussed in the comments). Since he's the original author and there might be a good reason behind this.

Reply:
I added this as a simple way to avoid collisions, without relying on the semantics of the code that invokes this.

I would try and think through what happens after a failure/recovery when an old task (running on a network-partitioned TM that has not yet cancelled) is still writing to the same file as a recovered task. Can that happen? If yes, what is the behavior then? Do we reduce the guarantees in that case? The partitioned TM should not be able to commit a checkpoint (to not roll over to a proper file), but can it mess up / overwrite the staging contents? My initial thought is that it probably could - not 100% sure though how HDFS leases would behave then, could they prevent that?